.github/workflows/deps-lock.yaml

```diff
@@ -0,0 +1,28 @@
+name: "Update deps-lock.json"
+on:
+  push:
+    paths:
+      - "deps.edn"
+
+permissions:
+  contents: write
+
+jobs:
+  update-lock:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.head_ref }}
+
+      - uses: cachix/install-nix-action@v25
+
+      - name: Update deps-lock
+        run: "nix run github:jlesquembre/clj-nix#deps-lock"
```

Comment:
Double-check with @ships to make sure this is still right, there's been some changes to the nix shells, since most of us are running direnv.